src/caf/core/data_structures.py

```diff
@@ -635,6 +670,12 @@ def aggregate(self, segs: list[str]):
         segs: Segments to aggregate to. Must be a subset of self.segmentation.naming_order,
         naming order will be preserved.
         """
+        if not isinstance(segs, list):
```

Comment:
Good points, this section is actually changed in another PR so I'm not going to address here, but yes that is handled lower down and in the newer change it also allows a new Segmentation instance to be passed, rather than a list of string segment names.